astropy/utils/data.py

```diff
@@ -232,6 +233,7 @@ def get_readable_fileobj(name_or_obj, encoding=None, cache=False,
         try:
             import bz2
         except ImportError:
+            fileobj.close()
```

Comment:
Yes, left over from the earlier attempt in tracking down the open-file errors; I guess it's not needed at all, but will check later on my notebook where psutil is installed.